src/Services/FHIR/FhirObservationService.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace OpenEMR\Services\FHIR;
+
+use OpenEMR\Services\BaseService;
+
+use OpenEMR\FHIR\R4\FHIRDomainResource\FHIRObservation;
+use OpenEMR\FHIR\R4\FHIRElement\FhirId;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRCodeableConcept;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRDateTime;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRReference;
+
+class FhirObservationService extends BaseService
+{
+    private $FHIRData;
+
+    public function __construct()
+    {
+        parent::__construct('form_vitals');
+        $this->FHIRData = array();
+    }
+
+    public function createObservationResource($id = '', $data = '', $encode = true)
+    {
+        switch ($data['formdir']) {
+            case 'vitals':
+                $coding['coding']['system'] = "http://terminology.hl7.org/CodeSystem/observation-category";
+                $coding['coding']['code'] = "vital-signs";
+                $coding['coding']['display'] = "Vital Signs";
+                $coding['text'] = "Vital Signs";
+                $this->FHIRData['category'] = ['category' => new FHIRCodeableConcept($coding)];
+                $this->addMembers($data);
+                break;
+            default:
+                #TODO: Return CapabilityStatement or Outcome Resource
+                break;
+        }
+        $this->FHIRData['subject'] = new FHIRReference(['reference' => "Patient/" . $data['pid'], "type" => "Patient"]);
+        $this->FHIRData['encounter'] = new FHIRReference(['reference' => "Encouter/" . $data['encounter'], "type" => "Encouter"]);
+        $this->FHIRData['performer']['performer'] = new FHIRReference(['reference' => "Practitioner/" . $data['provider_id'], "type" => "Practitioner"]);
+        $this->FHIRData['effectiveDateTime'] = new FHIRDateTime($data['date']);
```

Comment:
This is not the way we were dealing with other Resources. I just tried something new and this looks simple than the normal one. If anybody has any suggestion or want me to revert I am open :))